### PR TITLE
Enhancement/no auth interface

### PR DESCRIPTION
**Pull request by osala-eng**
## Description

Please include a summary of the change, e.g. what the new feature # is and/or what bug # it fixes. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes/Implements #(issue/feature)
Handle public no-auth sessions.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor and code improvement (non-breaking change which improves code quality and/or performance)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] Tests
- [ ] Other chores such as maintenance

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration by modifying the list below.

- [ ] Test A
- [ ] Test B
- [ ] Test C

**Test Configuration**:

Please describe the test setup. List them below as bullet points.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] The commit message follows the convention of this project

### CODE REVIEW

**Comment by osala-eng** [comment]
#### No auth Public Interface:

A js widget which can be embedded to any website. The front implementation  -> https://github.com/docqai/widget

#### Requirements
- There is a need for businesses to offer external individuals access to specific portions of their data, primarily content that is already publicly available, like product literature or a knowledge base.
- The proposed feature will make organizational data more versatile and not limited to just internal uses.

#### Proposed Solution
- [x] An embeddable JavaScript widget, which can be embedded in any website.
- [ ] Linking to ongoing projects (#14  and #22 )

#### Security Implementations
- [ ] A specific permission or role for making spaces public.
- [ ] An admin kill switch for quick disabling of space if a data leak is suspected.
- [ ] Requiring two different users to approve turning a space public.
- [ ] Guard rails in the future to warn users based on specific rules.

#### Screenshots

View with no public spaces configured
no_public_space

View with public spaces configured
first_look

sample_responses

**Comment by cwang** [comment]
Nice, a quick question: What if a private space is (mis-)configured for it? Is there gonna be an error message or something similar to the first screenshot? 

Also @osala-eng pls bring the branch up to date before I start reviewing it

File: web/utils/handlers.py [hunk 1]
```diff
@@ -408,3 +414,29 @@ def handle_get_gravatar_url() -> str:
     size, default, rating = 200, "identicon", "g"
     email_hash = hashlib.md5(email.lower().encode("utf-8")).hexdigest()  # noqa: S324
     return f"https://www.gravatar.com/avatar/{email_hash}?s={size}&d={default}&r={rating}"
+
+
+def handle_public_session() -> None:
+    """Handle public session."""
+    session_id = st.experimental_get_query_params().get("id", [None])[0]
```
Comment (janaka) [inline web/utils/handlers.py:421]: is this a chat session id being passed as a query string param? might be safer to use a less generic name here.

File: web/utils/handlers.py [hunk 2]
```diff
@@ -408,3 +414,29 @@ def handle_get_gravatar_url() -> str:
...
+    if session_id:
+        st.session_state["no_auth_public_session"] = True
```
Comment (janaka) [inline web/utils/handlers.py:423]: wondering if we should be setting session state directly like this as opposed to using one of the utils in session.py or adding utils if needed.

Comment (janaka) [reply web/utils/handlers.py:423]: those utils help ensure there aren't clashes by using key naming conventions.

Comment (janaka) [reply web/utils/handlers.py:423]: probably to add something like `ASK_PUBLIC` the enum `FeatureType` etc.

File: web/utils/layout.py [hunk 3]
```diff
@@ -710,7 +736,8 @@ def admin_docs_ui(q_param: str = None) -> None:
     spaces = list_shared_spaces()
 
     def _on_change() -> None:
-        del st.session_state["admin_docs_active_space"]
+        if "admin_docs_active_space" in st.session_state:
```
Comment (janaka) [inline web/utils/layout.py:739]: same as above might apply here also

File: web/index.py [hunk 4]
```diff
@@ -16,6 +16,7 @@
         Page("web/personal_chat.py", "General_Chat"),
         Page("web/personal_ask.py", "Ask_Your_Documents"),
         Page("web/personal_docs.py", "Manage_Your_Documents"),
+        Page("web/embed.py", "widget", "🔗"),
```
Comment (janaka) [inline web/index.py:19]: this renders a menu option right? is that basically a required hack?

File: web/utils/handlers.py (hunk 1 above)
Comment (osala-eng) [reply web/utils/handlers.py:421]: Okay, taking note of this.

File: web/index.py (hunk 4 above)
Comment (osala-eng) [reply web/index.py:19]: Yes the page has to registered in this manner although I made sure it is hidden in the menu.

File: web/utils/handlers.py (hunk 2 above)
Comment (osala-eng) [reply web/utils/handlers.py:423]: Okay, are you free for a quick call tomorrow on how to handle security implementations?

**Comment by osala-eng** [comment]
> Nice, a quick question: What if a private space is (mis-)configured for it? Is there gonna be an error message or something similar to the first screenshot?
> 
> Also @osala-eng pls bring the branch up to date before I start reviewing it

Currently works only with public spaces i.e. `ASK_SHARED` must be enabled and a space must have `PUBLIC` Permission on it.

**Comment by osala-eng** [comment]
> @osala-eng let us know when this branch is ready

Okay, I'm adding a method to clean up public session data files in the backend then it'll be ready.

**Review by janaka — APPROVED** [review]

**Comment by cwang** [comment]
This PR is problematic, primarily because it's implemented against a single space instead of a space group, with an uncessary introduction of a new space type.

**Pull request merged**